src/cascadia/TerminalControl/TermControlAutomationPeer.cpp

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "pch.h"
+#include <UIAutomationCore.h>
+#include "TermControlAutomationPeer.h"
+#include "TermControl.h"
+#include "TermControlAutomationPeer.g.cpp"
+
+#include "XamlUiaTextRange.h"
+
+namespace winrt::Microsoft::Terminal::TerminalControl::implementation
+{
+    TermControlAutomationPeer::TermControlAutomationPeer(winrt::Microsoft::Terminal::TerminalControl::implementation::TermControl const& owner) :
+        TermControlAutomationPeerT<TermControlAutomationPeer>(owner),
+        _uiaProvider{ owner.GetRenderData(), nullptr, std::bind(&TermControlAutomationPeer::GetBoundingRectWrapped, this) } {};
+
+    winrt::hstring TermControlAutomationPeer::GetClassNameCore() const
+    {
+        return L"TermControl";
+    }
+
+    AutomationControlType TermControlAutomationPeer::GetAutomationControlTypeCore() const
+    {
+        return AutomationControlType::Text;
+    }
+
+    winrt::hstring TermControlAutomationPeer::GetLocalizedControlTypeCore() const
+    {
+        // TODO GitHub #2142: Localize string
+        return L"TerminalControl";
+    }
+
+    winrt::Windows::Foundation::IInspectable TermControlAutomationPeer::GetPatternCore(PatternInterface patternInterface) const
+    {
+        switch (patternInterface)
+        {
+        case PatternInterface::Text:
+            return *this;
+            break;
+        default:
+            return nullptr;
+        }
+    }
+
+#pragma region ITextProvider
+    winrt::com_array<Windows::UI::Xaml::Automation::Provider::ITextRangeProvider> TermControlAutomationPeer::GetSelection()
+    {
+        SAFEARRAY* pReturnVal;
+        THROW_IF_FAILED(_uiaProvider.GetSelection(&pReturnVal));
+        return SafeArrayToComArray(pReturnVal);
+    }
+
+    winrt::com_array<Windows::UI::Xaml::Automation::Provider::ITextRangeProvider> TermControlAutomationPeer::GetVisibleRanges()

```

Comment:
we could probably use the `Windows::UI::Xaml::Automation::Provider` namespace in this file to make this easier to read

Reply:
Did you one better, added some namespace aliases to make it not ambiguous.